packages/knex/src/query/QueryBuilderHelper.ts

```diff
@@ -48,8 +48,13 @@ export class QueryBuilderHelper {
 
     if (prop?.customType?.convertToJSValueSQL) {
       const prefixed = this.prefix(field, true, true);
-      /* istanbul ignore next */
-      return this.knex.raw(prop.customType.convertToJSValueSQL!(prefixed, this.platform) + ' as ' + this.platform.quoteIdentifier(alias ?? prop.fieldNames[0]));
+      const valueSQL = prop.customType.convertToJSValueSQL!(prefixed, this.platform);
+
+      if (alias === null) {
+        return this.knex.raw(valueSQL);
+      }
+
+      return this.knex.raw(valueSQL + ' as ' + this.platform.quoteIdentifier(alias ?? prop.fieldNames[0]));
```

Comment:
let's keep the coverage ignore hint as it was to maintain branch coverage
```suggestion
      /* istanbul ignore next */
      return this.knex.raw(valueSQL + ' as ' + this.platform.quoteIdentifier(alias ?? prop.fieldNames[0]));
```